daq/gcp.py

```diff
@@ -56,8 +57,16 @@ def __init__(self, config, callback_handler):
         self._storage = storage.Client(project=self._project, credentials=self._credentials)
         self._ensure_report_bucket()
         self._config_callbacks = {}
+        self._logging = logging.Client(credentials=self._credentials, project=self._project)
+
         LOGGER.info('Connection initialized at %s', get_timestamp())
 
+    def get_logging_client(self):
+        """Gets the stackdriver client"""
+        if hasattr(self, "_client_name"):
```

Comment:
what is this check for?  When is _client_name not defined?  If it's in the fail-out case when gcp_cred is not defined, then you should just set it to None in the constructor, so it will at least exist as an attribute (makes the check cleaner).  using hasattr is somewhat of a red-flag use...